### PR TITLE
Added profile configuration parameter to ec2_sd_configs

### PR DESCRIPTION
**Pull request by bigkraig**
We use federated logins with rotating tokens so configuring Prometheus with tokens is a pain. This PR will allow us to use a profile with rotating tokens for local Prometheus testing.

I upgraded the aws-sdk-go package as part of this, since they improved session handling since it was vendor'd.

### CODE REVIEW

**Comment by bigkraig** [comment]
@beorn7 @fabxc any thoughts on this?

File: retrieval/discovery/ec2.go [hunk 1]
```diff
@@ -72,20 +72,22 @@ func init() {
...
+	profile  string
 	port     int
 }
 
 // NewEC2Discovery returns a new EC2Discovery which periodically refreshes its targets.
 func NewEC2Discovery(conf *config.EC2SDConfig) *EC2Discovery {
 	creds := credentials.NewStaticCredentials(conf.AccessKey, conf.SecretKey, "")
 	if conf.AccessKey == "" && conf.SecretKey == "" {
-		creds = defaults.DefaultChainCredentials
+		creds = nil
```
Comment (fabxc) [inline retrieval/discovery/ec2.go]: Does this have anything to do with the changes or does it relate to some changes in the updated vendoring?

**Comment by fabxc** [comment]
If it's a valid authentication mechanism and doesn't break any existing use case, this is okay with me.
Can you confirm that it doesn't break any other use case @bigkraig?

File: retrieval/discovery/ec2.go (hunk 1 above)
Comment (bigkraig) [inline retrieval/discovery/ec2.go]: It's related to the updated vendoring

**Comment by bigkraig** [comment]
Hey @fabxc, I just double checked that everything is still functioning with access_key & secret_key pairs. It looks good to me!

**Comment by fabxc** [comment]
Thanks for the clarification 👍 

My delayed response caused a merge conflict. I'll rebase and merge directly into master.

**Comment by tobstarr** [comment]
@bigkraig this change has nothing to do with IAMInstanceProfiles, right? Looks to me as if those are not supported at all at the moment, right?

**Comment by bigkraig** [comment]
@tobstarr right, this is just for STS. IAM instance profiles were already working, you should just leave all of the variables empty

**Comment by tobstarr** [comment]
@bigkraig Thanks for the heads-up. So also leave out the region?

I tried to set that up but nothing showed up in neither the targets, nor the logs. Will try to make the logs a bit more verbose.

**Comment by bigkraig** [comment]
I'm not sure about the region but I'd leave that in. You can always run the AWS CLI tools on an instance with the role and test things out too.

**Comment by joaogbcravo** [comment]
Hey! 

Is this supposed to work with credentials and config files on .aws folder on home directory?

I have this on .aws/config:
```
[profile default]
region = us-east-1

[profile super_role]
source_profile = default
role_arn = arn:aws:iam::XXX:role/Prometheus
region = us-east-1
```

I have on prometheus.yml
```
- job_name: node-ec2-us-east-1
  ec2_sd_configs:
    - region: us-east-1
      profile: super_role
      port: 9100
```

But I got this:
```
ERRO[0000] could not describe instances: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors  source="ec2.go:102"
```

**Comment by bigkraig** [comment]
@joaogbcravo yeah that is right, not sure why that isnt working. can you use that with a `--profile` parameter to `awscli`?

**Comment by joaogbcravo** [comment]
Hey!

Yes I can.
Something like 
```aws ec2 describe-instances --profile=super_role ```
works for me

**Pull request merged**